app/assets/javascripts/admin/forms/project_step_form.js.coffee

```diff
@@ -0,0 +1,151 @@
+$ ->
+  formatTypeOptions = (option) ->
+    icon = switch
+      when option.id == 'step' then 'flag'
+      when option.id == 'milestone' then 'calendar-check-o'
+    $('<i class="fa fa-' + icon + '"></i> <span>' + option.text + '</span>') unless !icon
+
+  addToAvailableLocales = (scope, locale) ->
+    elem = $(scope).find('#add_another_language')
+    $(elem).data('available-locales').splice(0, 0, locale)
+    setLinksVisibility(scope)
+
+  getNextAvailableLocale = (scope) ->
+    elem = $(scope).find('#add_another_language')
+    # hide the add new language link if we are returning the last unused locale    
+    locale = $(elem).data('available-locales').splice(0, 1)[0]
+    setLinksVisibility(scope)
+    locale
+
+  setLinksVisibility = (scope) ->
+    els = $(scope).find('.remove_language')
+    if els.length == 1
+      $(els).hide()
+    else
+      $(els).show()
+
+    el = $(scope).find('#add_another_language')
+    if $(el).data('available-locales').length == 0
+      $(el).hide()
+    else
+      $(el).show()
+
+  initTypeSelects = (scope) ->
+    $(scope).find(".type").select2({
+      theme: "bootstrap",
+      minimumResultsForSearch: Infinity,
+      width: "100%",
+      templateResult: formatTypeOptions,
+      templateSelection: formatTypeOptions
+    });
+
+  setCallbacks = (scope) ->
+
+    #
+    # show form
+    #
+   
+    $(scope).find('.edit-step-action').click (evt) ->
+      evt.preventDefault()
+      $(scope).find(".show-step-view").addClass('edit-step-view').removeClass('show-step-view')
```

Comment:
The actual method bodies like the above two lines should move into methods like `edit_step_clicked`